### PR TITLE
Custom metric thresholds

### PR DESCRIPTION
**Pull request by bgood**
Start of the PR to implement custom thresholds (Issue #9)

### CODE REVIEW

File: poller/README.md [hunk 1]
```diff
@@ -56,30 +60,55 @@ instructions on how to change the payload.
...
+
+To specify a custom threshold specify the name of the metrics to customize
+followed by the parameter values you wish to change. The updated parameters
+will be merged with the default metric parameters.
+
+### Selectors
+
+Key                  | Description
+---------------------| -----------
+`name`               | The unique name of the for the metric to be evaulated. The default metrics are `high_priority_cpu`, `rolling_24_hr` and `storage`.
```
Comment (davidcueva) [inline poller/README.md:98]: "**A** unique name ..."
It should be clear that this is a just a unique identifier that one can use for the metric. 

Naming the default metrics should be in the context of:
"In you want to override the default metrics, their names are: ...."

File: poller/index.js [hunk 2]
```diff
@@ -190,14 +190,28 @@ function postPubSubMessage(spanner, metrics) {
...
-      ...spanners[i],
-      ...await getSpannerMetadata(spanners[i].projectId, spanners[i].instanceId)
+    spanners[sIdx] = {...spannerDefaults, ...spanners[sIdx]};
+
+    spanners[sIdx].metrics =
+      buildMetrics(spanners[sIdx].projectId, spanners[sIdx].instanceId);
+    // merge in custom thresholds
+    if(metricOverrides != null) {
+      for (var oIdx = 0; oIdx < metricOverrides.length; oIdx++) {
+        mIdx = spanners[sIdx].metrics.findIndex(x => x.name === metricOverrides[oIdx].name);
```
Comment (davidcueva) [inline poller/index.js:204]: We could modify this loop to not only to override thresholds with custom ones but also to add custom metrics if the metric name is not found in the defaults.

File: poller/README.md (hunk 1 above)
Comment (bgood) [reply poller/README.md:98]: This implementation is pretty close to supporting custom metrics.  I think there are a couple things we should do in addition to this PR for custom metrics, added some thoughts to the Issue #10 around additional changes for custom metrics.

Comment (bgood) [reply poller/README.md]: README updated, per your recommendations.

File: poller/README.md [hunk 3]
```diff
@@ -56,30 +60,55 @@ instructions on how to change the payload.
...
+
+### Parameters
+
+Key                        | Description
+-------------------------- | -----------
+`filter`                   | The Cloud Monitoring metrics filter that should be used when querying for data.  This filter needs to include the project and instance_id filters.
```
[the remark this answers is not in the document]
Comment (bgood) [reply poller/README.md]: Great links thank you.  Updated the doc to include the links.

File: poller/index.js (hunk 2 above)
Comment (bgood) [reply poller/index.js:204]: Yes, that is where we'll add custom metrics.  I would like to make other changes to make custom metrics implementation cleaner so in the spirit of keeping this PR small and focused just on thresholds, I was planning on doing that work in separate PR.

**Review by davidcueva — APPROVED** [review]

**Pull request merged**